### DISCUSSION TITLE
Instructions to add support to Linux and macos to the Unreal Engine plugin

### DISCUSSION THREAD

**Discussion by demonixis** [JSBSim-Team/jsbsim#832] [General]
Hello everyone,
In the UE Plugin instructions it's indicated that the FSBSim library have to be compiled as dynamic. A VS solution is here on Windows to do the job and it's fine. How to do the same for Linux and Macos? I can edit the build.cs build script to add macos/linux lib support but I'm not sure how to generate them.
Thanks,
Yannick.

**Comment by gallonmate** [comment]
There's already been a discussion started but with no solution yet. https://github.com/JSBSim-Team/jsbsim/discussions/791#discussion-4692945

**Comment by demonixis** [comment]
Thanks for the link. What I want is more how to create a dynamic library for macos and Linux. I don't see anything in the docs that mention how to do that with cmake for instance.
If we can build a dynamic library, we can create a script that does the same things done by the JSBSimUnreal C++ solution. Finally we just have to edit one file to link this library with the Unreal build system.

But first thing to do is to find how to do a dynamic library for macos and Linux.

**Reply by bcoconni** [reply]
> Thanks for the link. What I want is more how to create a dynamic library for macos and Linux. I don't see anything in the docs that mention how to do that with cmake for instance.

JSBSim is using [CMake](https://cmake.org/) to build on both Linux and MacOS. So you can compile JSBSim as a dynamic library using [CMake standard flag `BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html?highlight=build_shared_libs#variable:BUILD_SHARED_LIBS):
```bash
> mkdir build
> cd build
> cmake -DBUILD_SHARED_LIBS=ON ..
> make
```

**Reply by demonixis** [reply]
Thanks, I tried to build using this command and it worked, I've the following result
- JSBSIm.Framework
- JSBSim (exec)

The JSBSim.Framework folder contains only a link to JSBSim with an info.plist. There is no libraries in it.
It's probably linked elsewhere. My next question is: "How to build the JSBSim.Framework with everything in it to be able to use it in another location? 

Or maybe I can just do make install and use that system path in the Unreal build system?

**Comment by herewego48** [comment]
Have you successfully added the jsbsim plug-in to UE5? I simply copy the JSBSimFlightDynamicsModel folder to the plugins folder of UE5. UE5 can find this plugin, but prompts me to compile it.
